pkg/controller/operators/decorators/operator.go

```diff
@@ -122,24 +182,51 @@ func (a *Operator) ComponentSelector() (labels.Selector, error) {
 }
 
 // ResetComponents resets the component selector and references in the operator's status.
-func (a *Operator) ResetComponents() error {
-	labelSelector, err := a.ComponentLabelSelector()
+func (o *Operator) ResetComponents() error {
+	labelSelector, err := o.ComponentLabelSelector()
 	if err != nil {
 		return err
 	}
 
-	a.Status.Components = &operatorsv2alpha1.Components{
+	o.Status.Components = &operatorsv2alpha1.Components{
 		LabelSelector: labelSelector,
 	}
 
 	return nil
 }
 
+// AdoptComponent adds the operator's component label to the given component, returning true if the
+// component label was added and false if it already existed.
+func (o *Operator) AdoptComponent(component runtime.Object) (adopted bool, err error) {
+	var labelKey string
+	if labelKey, err = o.ComponentLabelKey(); err != nil {
+		return
+	}
+
+	var m metav1.Object
+	if m, err = meta.Accessor(component); err != nil {
+		return
+	}
+
+	labels := m.GetLabels()
+	if labels == nil {
+		labels = map[string]string{}
+		m.SetLabels(labels)
+	}
+
+	if _, ok := labels[labelKey]; !ok {
+		labels[labelKey] = ""
```

Comment:
nit: this is just a preference, but if we are adding a `""` to signify that a label is present, is it more intuitive to use an annotation instead?

Reply:
annotations are also key value pairs, is there another reason why an annotation is more intuitive?